prisma/schema.prisma

```diff
@@ -10,8 +10,145 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
+model Gender {
+  id           String   @id @default(uuid())
+  abbreviation String   @unique
+  description  String
+  User         User[]
+  createdAt    DateTime @default(now()) @db.Timestamptz()
+}
+
 model User {
-    id String @id @db.Uuid()
-    name String 
-    createdAt DateTime @default(now())
+  id               String             @id @default(uuid())
```

Comment:
I followed the official prisma docs, pretty sure that's mapped by default, and it's `db.uuid` not `db.uuid()`
![image](https://github.com/chingu-x/chingu-dashboard-be/assets/6191116/8fc898b5-3bc2-4c9e-b186-2edcb2ace5bf)
![image](https://github.com/chingu-x/chingu-dashboard-be/assets/6191116/8651df7d-fd9e-4291-b8d7-1d334b94461e)

Reply:
Ok, the type should be `@dbUuid` inside the prisma model, I think it is important that the type in the database should be of Uuid, more descriptive of what it is and will have a degree of validation at the database level

Reply:
url for your reference: https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#generate-uuid-values-as-ids

Reply:
I think it might be easier to just utilize autoincrement id instead of uuid.

Reply:
i think so too. what's the advantage of working with uuid? 95% of the prisma/postgres examples I see use autoincrement id

Reply:
What I am talking about is using the correct type in postgres, adding the default type will only create a uuid value on insert, but it could be overriden by anyone by a raw query, what I am suggesting will add a uuid type in the database, which will force to use always a uuid value in there

![Screenshot 2023-09-16 at 11 30 58](https://github.com/chingu-x/chingu-dashboard-be/assets/97707578/a4fd8415-2c36-4943-bedd-cda8da0cc4fd)

url: https://www.prisma.io/docs/concepts/database-connectors/postgresql#native-type-mappings

How it is written now makes the type in the database to be text

![Screenshot 2023-09-16 at 11 39 03](https://github.com/chingu-x/chingu-dashboard-be/assets/97707578/1f1a7fbc-dff4-4d97-8a63-5991cce32a88)

Reply:
The change will look like this in the prisma file 
![Screenshot 2023-09-16 at 11 41 22](https://github.com/chingu-x/chingu-dashboard-be/assets/97707578/49f29fb7-ef17-40d2-9bb3-d0a81437755c)

and that will result in the following in the database
![Screenshot 2023-09-16 at 11 42 03](https://github.com/chingu-x/chingu-dashboard-be/assets/97707578/b4445f77-3d4e-4b82-b222-7e86ad864599)

Reply:
> I think it might be easier to just utilize autoincrement id instead of uuid.

The change is equally as hard if changing to autoincrements or adding the uuid type to the database

Reply:
> The change will look like this in the prisma file ![Screenshot 2023-09-16 at 11 41 22](https://user-images.githubusercontent.com/97707578/268459903-49f29fb7-ef17-40d2-9bb3-d0a81437755c.png)
> 
> and that will result in the following in the database ![Screenshot 2023-09-16 at 11 42 03](https://user-images.githubusercontent.com/97707578/268459923-b4445f77-3d4e-4b82-b222-7e86ad864599.png)
> 
> In the example above you can see that the id has the type of uuid in the database

I know what @db.Uuid does but it doesn't answer my questions. Why is having UUID type in postgres itself necessary? Will it make things more complicated when using prisma? What's the advantage of not using what prisma does things on their docs? And I also think we don't need uuid for most of the tables

Reply:
Sorry misunderstood your question, for Prisma there will not be any difference on the type we use, it will give us the correct type where we call it. What it does is that it adds a layer of validation of the data at the table level, so we know for certain that all the values in the id column are valid uuid's.  I'd rather have a standard across the db on what our id type will be, but if you see fit using an autoincrement in a specific table, in some cases it might be a good idea.  The only db engine that I know of that uses the `uuid` type is postgres, and that is why I believe that in their documentation Prisma uses a `string`

Reply:
it only matters if we will ever insert directly without using prisma, that's the point of using prisma.. but i'm not even sure if we need uuid for everything. do you really want the endpoints be like `/voyages/b0e11958-e058-4da4-b5fd-0d17c4ab2ce0/member/b0e11958-e058-4da4-b5fd-0d17c4ab2ce0/`

Reply:
remember the member id will be dropped when we have authentication, however I don't mind having those url's, and as I said in some cases it might be a good idea to use another kind of way on setting up the id like autoincrement

Reply:
> member id will be dropped when we have authentication
what do you mean by this?

Reply:
The second Id is of the logged in user, that was one of the things I remember talking about today, and it is that we should read the user info from the auth cookies, so we keep that endpoint until then so we have that info, and the reason I've always talked about implementing the OAuth part as soon as possible

Reply:
What are your thoughts on this @marktlinn. I'd like to get this resolved, like now (asap).

Reply:
mark responded on discord, resolved.